### PR TITLE
add pool + bond queries, rotate

### PR DESCRIPTION
**Pull request by sirlensalot**

### CODE REVIEW

File: pact/relay/pool.pact [hunk 1]
```diff
@@ -90,6 +104,17 @@
   (defcap ACTIVITY ( pool:string bond:string activity:integer)
     @event true)
 
+
+  (defcap ROTATE (bond:string account:string)
```
Comment (ggobugi27) [inline pact/relay/pool.pact:108]: We can take out `account` from the parameter. Or, is it intended to so people know what they're signing for?

Comment (sirlensalot) [reply pact/relay/pool.pact:108]: Yeah, it's doubling as an event

**Review by ggobugi27 — CHANGES_REQUESTED** [review]
Rotating the account was unexpected. Do we really need this feature? Why don't we only rotate the keys? This could act as a `TRANSFER` of the amounts without actually calling `TRANSFER`. 

Some scenarios that could happen would be 
i.e. Alice could lose the bond amount by sending in `(relay.pool.rotate "Alice:2021-01-01" "bob-fake" (read-keyset 'bob))`, and the bond is `bob-fake`'s, and having bob's key on the bond won't help bringing it back.

File: pact/relay/pool.pact (hunk 1 above)
Comment (ggobugi27) [reply pact/relay/pool.pact:108]: Why don't we use different name for `account` in  line 112 and line 115? Something like orig-acct ? Just to be clear about which guard we're enforcing.

**Comment by sirlensalot** [comment]
The goal was to allow us to address possibly implementing payment flows to operators, e.g. maybe directing activity fees to operators and APY to bonding-only bonders. I agree it's too heavy-handed and changed it to just rotating the key.

File: pact/relay/pool.pact [hunk 2]
```diff
@@ -104,16 +104,9 @@
...
-    "Rotation of bond credentials requires auth from originating account."
+  (defcap ROTATE ( bond:string )
     @managed
-    (with-read bonds bond
-      { 'pool := pool, 'account := account }
-      (with-read pools pool
-        { 'token := token:module{fungible-v2} }
-          (enforce-guard (at 'guard (token::details account)))))
-  )
+    (enforce-guard (at 'guard (get-bond bond))))
```
Comment (ggobugi27) [inline pact/relay/pool.pact:109]: I thought it was reasonable to enforce the account guard when rotating the bond guard. Was thinking of bond key as a kind of sub-key that can be used in the relay software. Don't have a strong opinion on it though.

Comment (sirlensalot) [reply pact/relay/pool.pact:109]: That makes sense

**Review by ggobugi27 — APPROVED** [review]
Looks good to me!

**Pull request merged**